File: frontend/src/app/shared/models/items/weapon-type.ts
import {Attribute} from '../attributes/attribute';
import {AttributeType} from '../attributes/attribute-type.enum';

export class WeaponType extends Attribute {
  type = 'WeaponType';

  constructor() {
    super();
    this.attributeType = AttributeType.WEAPON_TYPE;
  }
}

File: backend/src/main/java/com/herd/squire/models/attributes/WeaponType.java
package com.herd.squire.models.attributes;

public class WeaponType extends Attribute {
    public WeaponType() {
    }

    public WeaponType(String id, String name, String description, int sid, boolean author, int version) {
        super(id, name, description, AttributeType.WEAPON_TYPE, sid, author, version);
    }
}
